packages/provider/src/wallet.ts

```diff
@@ -619,7 +617,9 @@ export class Wallet implements WalletProvider {
 
     // an extra override for convenience
     if (this.config.networkRpcUrl) {
-      this.networks[0].rpcUrl = this.config.networkRpcUrl
+      // TODO: Is this okay? why do we need to overwride the rpc here?
```

Comment:
we allow the dapp to override it they can have preference over the node-provider, as well if they are a very high usage dapp. In general, the user could change the rpcUrl on the wallet side itself, however, as a Dapp developer who uses 0xsequence directly, to make it easy for them and their users, we allow the dapp to override the rpcUrl here as well